File: javascript/1704-halvesAreAlike.js
// https://leetcode.com/problems/determine-if-string-halves-are-alike/
/**
 * @param {string} s
 * @return {boolean}
 */
 var halvesAreAlike = function(s) {
  let f1 = s.substring(0, s.length/2);
  let f2 = s.substring(s.length/2, s.length);

  let alike = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];

  let p1 = "";
  let p2 = "";

  for(let i=0; i < f1.length; i++) {
      if(alike.includes(f1.charAt(i))) p1 += f1.charAt(i);
  }

  for(let i=f2.length-1; i >= 0; i--) {
      if(alike.includes(f2.charAt(i))) p2 += f2.charAt(i);
  }

  console.log(f1, p1, 'first part');
  console.log(f2, p2, 'second part');

  if(p1.length == p2.length) return true;

  return false;
};

File: java/1704-halvesAreAlike.java
class Solution {
  public boolean halvesAreAlike(String s) {
      String a = s.substring(0, s.length()/2);
      String b = s.substring(s.length()/2, s.length());

      System.out.println(a);
      System.out.println(b);

      String[] alike = {"a", "e", "i", "o", "u", "A", "E", "I", "O", "U"};

      String p1 = "";
      for(int i=0; i < a.length(); i++) {
          if(Arrays.asList(alike).contains(a.substring(i, i+1))) {
              p1 += a.substring(i, i+1);
          }
      }

      System.out.println(p1);

      String p2 = "";
      for(int i=0; i < b.length(); i++) {
          if(Arrays.asList(alike).contains(b.substring(i, i+1))) {
              p2 += b.substring(i, i+1);
          }
      }

      System.out.println(p2);

      if(p1.length() == p2.length())
          return true;

      return false;
  }
}
